File: ships2.py
ships = {}
while True:
    ship = input('Введите название корабля: ')
    if ship == '':
        break
    if ship not in ships:
        ships[ship] = 1
    else:
        if len(ship) < 6:
            ships[ship] += 1
        else:
            ships[ship] *= 2
print(ships)

File: lua/Ships2.lua
ships = {}
while true do
    io.write("Введите название корабля: ")
    local ship = io.read()
    if ship == "" then
        break
    end
    if ships[ship] == nil then
        ships[ship] = 1
    else
        ships[ship] = ships[ship] + 1
    end
    for k,v in pairs(ships) do
        print(k.. ":" ..v)
    end
end
